src/components/Chat/hooks/useChat.ts

```diff
@@ -0,0 +1,139 @@
+import { useCallback, useEffect, useState } from 'react';
+import Dayjs from 'dayjs';
+
+import { Streami18n } from '../../../i18n';
+
+import { version } from '../../../../package.json';
+
+import type { Channel, Event, Mute, StreamChat } from 'stream-chat';
+import type { TranslationContextValue } from '../../../context/TranslationContext';
+import type {
+  DefaultAttachmentType,
+  DefaultChannelType,
+  DefaultCommandType,
+  DefaultEventType,
+  DefaultMessageType,
+  DefaultReactionType,
+  DefaultUserType,
+  UnknownType,
+} from '../../../../types/types';
+
+export type UseChatParams<
+  At extends UnknownType = DefaultAttachmentType,
+  Ch extends UnknownType = DefaultChannelType,
+  Co extends string = DefaultCommandType,
+  Ev extends UnknownType = DefaultEventType,
+  Me extends UnknownType = DefaultMessageType,
+  Re extends UnknownType = DefaultReactionType,
+  Us extends UnknownType = DefaultUserType
+> = {
+  client: StreamChat<At, Ch, Co, Ev, Me, Re, Us>;
+  i18nInstance?: Streami18n;
+  initialNavOpen?: boolean;
+};
+
+export const useChat = <
+  At extends UnknownType = DefaultAttachmentType,
+  Ch extends UnknownType = DefaultChannelType,
+  Co extends string = DefaultCommandType,
+  Ev extends UnknownType = DefaultEventType,
+  Me extends UnknownType = DefaultMessageType,
+  Re extends UnknownType = DefaultReactionType,
+  Us extends DefaultUserType<Us> = DefaultUserType
+>({
+  client,
+  i18nInstance,
+  initialNavOpen,
+}: UseChatParams<At, Ch, Co, Ev, Me, Re, Us>) => {
+  const [translators, setTranslators] = useState<TranslationContextValue>({
+    t: (key) => key,
+    tDateTimeParser: (input) => Dayjs(input),
+    userLanguage: 'en',
+  });
+
+  const [channel, setChannel] = useState<
+    Channel<At, Ch, Co, Ev, Me, Re, Us> | undefined
+  >(undefined);
```

Comment:
should work the same
```suggestion
  const [channel, setChannel] = useState<Channel<At, Ch, Co, Ev, Me, Re, Us>>();
```